pmd-java/src/test/resources/net/sourceforge/pmd/lang/java/rule/codestyle/xml/VariableCanBeInlined.xml

```diff
@@ -0,0 +1,1010 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<test-data
+        xmlns="http://pmd.sourceforge.net/rule-tests"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://pmd.sourceforge.net/rule-tests http://pmd.sourceforge.net/rule-tests_1_0_0.xsd">
+    <!--        -->
+    <!-- return -->
+    <!--        -->
+    <test-code>
+        <description>skip void/native/abstract methods</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public void bar() {}
+    public native int buz();
+    public abstract int baz();
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>skip literal returns</description>
+        <expected-problems>0</expected-problems>
+        <code><![CDATA[
+public class Foo {
+    public int bar() { return 5; }
+}
+        ]]></code>
+    </test-code>
+
+    <test-code>
+        <description>simple failure case</description>
+        <expected-problems>1</expected-problems>
```

Comment:
```suggestion
        <expected-problems>1</expected-problems>
        <expected-linenumbers>3</expected-linenumbers>
        <expected-messages>
            <message>Consider simply using the value vs storing it in local variable 'x'.</message>
        </expected-messages>
```

I want to make sure, that we report the correct line (now and in the future). We report the variable declaration, correct? Or do we report the return statement?

Yes, according to the implementation, we report on the node, where the local var has been declared.